clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -554,7 +555,7 @@ public void commitAsync() {
      */
     @Override
     public void setAcknowledgementCommitCallback(final AcknowledgementCommitCallback callback) {
-        throw new UnsupportedOperationException();
+        acknowledgementCommitCallbackHandler = new AcknowledgementCommitCallbackHandler(callback);
```

Comment:
I think you should use the `acquireAndEnsureOpen` logic in a try-finally block here too. Just to ensure single-threaded access to the consumer which is the convention we are using.